python/spear/instance.py

```diff
@@ -0,0 +1,236 @@
+        self._initialize_unreal_instance()
+
+    def close(self):
+        # Note that in the constructor, we launch the Unreal instance first and then initialize the RPC client. Normally,
+        # we would do things in the reverse order here. But if we close the client first, then we can't send a command to
+        # the Unreal instance to close it. So we close the Unreal instance first and then close the client.
+        self._request_close_unreal_instance()
+        self._close_rpc_client()
+
+    def _request_launch_unreal_instance(self):
+
+        if self._config.SPEAR.INSTANCE.LAUNCH_MODE == "none":
+            spear.log('SPEAR.INSTANCE.LAUNCH_MODE == "none" so we assume that an Unreal instance has been launched externally...')
+            return
+
+        spear.log("Launching Unreal instance...")
+
+        # write temp file
+        temp_dir = os.path.realpath(os.path.join(self._config.SPEAR.INSTANCE.TEMP_DIR))
+        temp_config_file = os.path.realpath(os.path.join(temp_dir, self._config.SPEAR.INSTANCE.TEMP_CONFIG_FILE))
+
+        spear.log("Writing temp config file: " + temp_config_file)
+
+        os.makedirs(temp_dir, exist_ok=True)
+        with open(temp_config_file, "w") as output:
+            self._config.dump(stream=output, default_flow_style=False)
+
+        # set up launch executable and command-line arguments
+        launch_args = []
+
+        if self._config.SPEAR.INSTANCE.LAUNCH_MODE == "uproject":
+            launch_executable = self._config.SPEAR.INSTANCE.EDITOR_EXECUTABLE
+            launch_args.append(self._config.SPEAR.INSTANCE.UPROJECT)
+            launch_args.append("-game") # launch the game using uncooked content
+        elif self._config.SPEAR.INSTANCE.LAUNCH_MODE == "standalone":
+            launch_executable = self._config.SPEAR.INSTANCE.STANDALONE
+        else:
+            assert False
+
+        assert os.path.exists(launch_executable)
+
+        launch_executable_name, launch_executable_ext = os.path.splitext(launch_executable)
+
+        if sys.platform == "win32":
+            assert launch_executable_name[-4:] == "-Cmd"
+            assert launch_executable_ext == ".exe"
+            launch_executable_internal = launch_executable
+        elif sys.platform == "darwin":
+            assert launch_executable_ext == ".app"
+            launch_executable_internal = os.path.realpath(os.path.join(launch_executable, "Contents", "MacOS", os.path.basename(launch_executable_name)))
+        elif sys.platform == "linux":
+            assert launch_executable_ext == ".sh"
+            launch_executable_internal = launch_executable
+        else:
+            assert False
+
+        assert os.path.exists(launch_executable_internal)
+
+        for arg, value in (self._config.SPEAR.INSTANCE.COMMAND_LINE_ARGS).items():
+            if value == None:
+                launch_args.append("-{}".format(arg))
+            elif isinstance(value, bool) and value == False:
```

Comment:
I don't think this `bool` special case should be here. Suppose I pass in the following:

```
COMMAND_LINE_ARGS:
  enable_slow_feature_that_is_turned_on_by_default: False
```
I expect that to be forwarded to Unreal as `-enable_slow_feature_that_is_turned_on_by_default=False`. I don't want it to be omitted.

Also, you should just say `not value` or `value` instead of `value == False` or `value == True`.

Reply:
You are right, we don't need this special case here. Removed it.